From 8c58ebad90340ad3ab5c54eee9a5db2bb46b363f Mon Sep 17 00:00:00 2001
From: Hans Christian Lonstad <hcl@datarespons.no>
Date: Thu, 28 Mar 2019 15:20:13 +0100
Subject: [PATCH 3/6] USB storage: Reduce max blocks in transfer

Some memory sticks chokes on big transfers - for instance USB 2.0 type
Kingston DataTraveler
---
 common/usb_storage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/usb_storage.c b/common/usb_storage.c
index 9cd64744f8..3ca348545f 100644
--- a/common/usb_storage.c
+++ b/common/usb_storage.c
@@ -952,7 +952,7 @@ static void usb_stor_set_max_xfer_blk(struct usb_device *udev,
 	 * there is enough free heap space left, but the SCSI READ(10) and
 	 * WRITE(10) commands are limited to 65535 blocks.
 	 */
-	blk = USHRT_MAX;
+	blk = 64;
 #else
 	blk = 20;
 #endif
-- 
2.20.1

